src/main/java/uk/gov/hmcts/reform/sscs/factory/NotificationFactory.java

```diff
@@ -35,18 +44,28 @@
             return null;
         }
 
-        Benefit benefit = getBenefitByCode(notificationWrapper.getSscsCaseDataWrapper().getNewSscsCaseData().getAppeal().getBenefitType().getCode());
-        Template template = personalisation.getTemplate(notificationWrapper, benefit);
+        Benefit benefit = getBenefitByCode(notificationWrapper
+                .getSscsCaseDataWrapper().getNewSscsCaseData().getAppeal().getBenefitType().getCode());
+        Template template = personalisation.getTemplate(notificationWrapper, benefit, subscriptionType);
 
         SscsCaseData ccdResponse = notificationWrapper.getSscsCaseDataWrapper().getNewSscsCaseData();
 
-        Destination destination = getDestination(ccdResponse.getSubscriptions().getAppellantSubscription());
+        Subscription subscription = getSubscriptionGivenSubscriptionType(ccdResponse, subscriptionType);
+        Destination destination = getDestination(subscription);
         Reference reference = new Reference(ccdResponse.getCaseReference());
-        String appealNumber = ccdResponse.getSubscriptions().getAppellantSubscription().getTya();
+        String appealNumber = subscription.getTya();
 
         return new Notification(template, destination, placeholders, reference, appealNumber);
     }
 
+    private Subscription getSubscriptionGivenSubscriptionType(SscsCaseData ccdResponse,
+                                                              SubscriptionType subscriptionType) {
+        if (APPELLANT.equals(subscriptionType)) {
+            return ccdResponse.getSubscriptions().getAppellantSubscription();
+        }
+        return ccdResponse.getSubscriptions().getRepresentativeSubscription();
```

Comment:
Need to add a check to verify representative subscription exists or not.

Reply:
No need @sathishbabuj , as explained above the subscriptions is always built by the deserialiser.